source/core/ut_suite_builder.pkb

```diff
@@ -768,26 +807,36 @@ create or replace package body ut_suite_builder is
     l_default_context_name t_object_name;
     function get_context_name(
       a_parent in out nocopy ut_suite,
-      a_context_names in tt_annotation_texts,
-      a_start_position binary_integer,
-      a_end_position binary_integer
+      a_start_position binary_integer
     ) return varchar2 is
       l_result         t_annotation_name;
       l_found          boolean;
+      l_end_position   binary_integer;
       l_annotation_pos binary_integer;
+      l_context_names  tt_annotation_texts;
     begin
-      l_annotation_pos := a_context_names.first;
-      while l_annotation_pos is not null loop
-        if l_annotation_pos > a_start_position and l_annotation_pos < a_end_position then
-          if l_found then
-            add_annotation_ignored_warning(a_parent, gc_name,'Duplicate annotation %%%.', l_annotation_pos);
-          else
-            l_result := a_context_names(l_annotation_pos);
+      if a_annotations.by_name.exists(gc_name) then
```

Comment:
This looks really great.
Nice improvement to code readability and functionality.